externaltask-event-listener.js

```diff
@@ -16,65 +16,34 @@ module.exports = function (RED) {
 
             let subscription;
 
+            function externalTaskCallback(event) {
+                return (externalTaskNotification) => {
+                    if (config.externaltask != '' && config.externaltask != externalTaskNotification.flowNodeId) return;
+                    node.send({
+                        payload: {
+                            flowNodeInstanceId: externalTaskNotification.flowNodeInstanceId,
+                            externalTaskEvent: externalTaskNotification,
+                            action: event,
+                            type: 'externaltask',
+                        },
+                    });
+                };
+            }
+
             async function subscribe() {
                 switch (config.eventtype) {
```

Comment:
Jetzt geht doch sogar der ganze switch block weg.
config.eventtype ist ja immer das was in externalTaskCallback rein kommt.

Ich würde den callback vielleicht auch umdrehen also:

return await client.notification.onExternalTaskCreated((externalTaskNotification) => externalTaskCallback(config.eventtype), {
                            identity: currentIdentity,
                        });

Reply:
Ich hab das switch statement jetzt rausgenommen. Ich bin mir aber nicht sicher ob meine neue Lösung besser oder schlechter zu verstehen und warten ist.

Was genau meinst du mit umdrehen? Den zweiten Teil verstehe ich noch nicht ganz.